### ISSUE TITLE
Decoder using less VRAM

### ISSUE THREAD

**Issue by Cortexelus** [descriptinc/descript-audio-codec#10]
I notice the decoder uses a ton of VRAM (and in some cases causing OOM error)

Here's me decoding a ~5 minute audio file
<img width="327" alt="image" src="https://github.com/descriptinc/descript-audio-codec/assets/2453861/dcd94415-5717-4bc2-a5a6-46786bf86484">

One solution is reduce VRAM by splitting the codes into chunks, decoding each chunk, then concatenating them
```python
# split into 100 chunks
chunks = np.array_split(data["codes"], 100, axis=-1)
final = []
for chunk in chunks:
    data["codes"] = chunk
    # decode each chunk
    recon = decode(data, 'cuda', model, preserve_sample_rate=True)
    final.append(recon.audio_data)
# concatenate audio data
concatenated = torch.cat(final, dim=-1)
# shorten the end to the original length
concatenated = concatenated[:,:,:data["metadata"]["original_length"]]
# save
torchaudio.save('recon.wav', concatenated[0], data["metadata"]["original_sr"])
```

However this causes audio normalization weirdness. You can see here, bottom to top (the original wav, default decoding (no chunking), 10 chunks, 160 chunks):
<img width="1013" alt="image" src="https://github.com/descriptinc/descript-audio-codec/assets/2453861/87b014ab-bcb0-44d8-a30e-3ed45fc5c671">

and clipping
<img width="435" alt="image" src="https://github.com/descriptinc/descript-audio-codec/assets/2453861/e8de4b82-21a9-4293-b685-a8551918689e">

How do I solve that

**Comment by zqevans** [comment]
I'm getting a lot of VRAM usage as well, it seems to largely be due to the `alpha*x` equation in the snake activation: 
https://github.com/descriptinc/descript-audio-codec/blob/06e80492f169d50cf1911d80d58ab16c2693de2e/dac/nn/layers.py#L22

**Comment by pseeth** [comment]
We'll have better chunking code soon! Just working through the logic to avoid clicks in the output. Sorry about that. Will close this when it's merged.

**Comment by pseeth** [comment]
This is now merged in the most recent DAC 1.0.0. Let me know if you have any issues! Use the `model.compress` and `model.decompress` functions.
